custom/enikshay/reports/views.py

```diff
@@ -100,6 +90,29 @@ def get(self, request, domain, case_type, *args, **kwargs):
         }
         return render(request, 'enikshay/duplicate_ids_report.html', context)
 
+    def add_user_info_to_cases(self, bad_cases):
+        user_info = self.get_user_info(
+            case['form_user_id'] for case in bad_cases if 'form_user_id' in case)
+
+        auth_user_ids = [case['auth_user_id'] for case in bad_cases
+                         if 'auth_user_id' in case]
+        auth_usernames = [user_doc['username'] for user_doc in
+                          iter_docs(CommCareUser.get_db(), auth_user_ids)]
+        for case in bad_cases:
+            user_dict = user_info.get(case.get('form_user_id'))
+            if user_dict:
+                case['username'] = user_dict['username']
+                device_id = case['form_device_id']
+                if device_id == 'Formplayer':
```

Comment:
huh, does formplayer use different device ids for submissions / restores? I thought they were of the form `WebAppsLogin|{username}|as|{user}`. (which is kind of what you are creating when checking this value here, but with * instead of |).

Reply:
Formplayer does use different device IDs, unfortunately.  It used to be `|`, but that's been switched to `*`, I think because some place where that gets serialized plays nicer with `*`

Reply:
huh, really? this doesn't actually break anything, but this testcase is no longer indicative of reality: 
https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/ota/tests/test_app_aware_restore.py#L126

Reply:
Yeah, and unfortunately making that switch essentially doubled the number of devices for eNikshay users (we keep track of those serially for ID generation)